test/e2e/dns.go

```diff
@@ -0,0 +1,409 @@
+package e2e
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"regexp"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/Azure/go-autorest/autorest/to"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/network"
+	"github.com/Azure/ARO-RP/pkg/util/stringutils"
+	"github.com/Azure/ARO-RP/test/util/project"
+)
+
+var (
+	nameserverRegex              = regexp.MustCompile("nameserver [0-9.]*")
+	verifyResolvConfTimeout      = 30 * time.Second
+	verifyResolvConfPollInterval = 1 * time.Second
+	nodesReadyPollInterval       = 2 * time.Second
+	nicUpdateWaitTime            = 5 * time.Second
+)
+
+const (
+	maxObjNameLen           = 63
+	resolvConfContainerName = "read-resolv-conf"
+)
+
+var _ = Describe("ARO cluster DNS", func() {
+	It("must not be adversely affected by Azure host servicing", func(ctx context.Context) {
+		By("creating a test namespace")
+		testNamespace := fmt.Sprintf("test-e2e-%d", GinkgoParallelProcess())
```

Comment:
Not required for this PR, but I think it would be an improvement to use this convention everywhere we create a test namespace in e2e. I see there are a few more examples of us creating test namespaces in the e2e suite.

Reply:
Oh, I took this from `cluster.go`, which I thought was the only other place where we create a test namespace. Where else did you see us creating a test namespace?

Reply:
This is what I found searching the code base during my last review: https://github.com/Azure/ARO-RP/blob/master/test/e2e/adminapi_cluster_getlogs.go#L35

But looking again, it doesn't look like we ever create that namespace - just confirm that we can't get logs from a pod there if we make an attempt. I think it's fine as-is!